### PR TITLE
[Merged by Bors] - feat(GroupTheory/FixedPointFree): New file

### PR DESCRIPTION
**Pull request by tb65536**
This PR adds a new file containing the definition of fixed-point-free automorphisms and some basic properties.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/List/BigOperators/Basic.lean
Comment (jcommelin) [inline Mathlib/Data/List/BigOperators/Basic.lean:452]: This seems very similar to `Finset.prod_range_div`. Do you think you can use that instead?

File: Mathlib/GroupTheory/FixedPointFree.lean
Comment (jcommelin) [inline Mathlib/GroupTheory/FixedPointFree.lean:30]: Does the `commutatorMap` deserve to be factored into a standalone definition?

File: Mathlib/GroupTheory/FixedPointFree.lean
Comment (jcommelin) [inline Mathlib/GroupTheory/FixedPointFree.lean:62]: Could you please add, for convenience, the corollaries that `G` is abelian, and has odd order (or no elements of order 2, or both)?

File: Mathlib/Data/List/BigOperators/Basic.lean
Comment (tb65536) [reply Mathlib/Data/List/BigOperators/Basic.lean:452]: `Finset.prod_range_div` assume `CommGroup`, but I suppose I could change this lemma to use `/`.

Comment (tb65536) [reply Mathlib/Data/List/BigOperators/Basic.lean:452]: Well actually, this lemma isn't quite the right way around to use `/`.

File: Mathlib/GroupTheory/FixedPointFree.lean
Comment (jcommelin) [inline Mathlib/GroupTheory/FixedPointFree.lean:21]: I think this should not be in the `MulEquiv` namespace. But I'm not sure in which namespace it *should* be... `Function` is bad, because in that context the function really not have any fixed points. `MulHom`? `MonoidHom`?
Maybe we should ask on zulip.

Comment (tb65536) [reply Mathlib/GroupTheory/FixedPointFree.lean]: `MonoidHom` seems most natural to me (the `MulEquiv` was just a holdover from when I had to state several of the theorems in terms of `MulEquiv`)

Comment (tb65536) [reply Mathlib/GroupTheory/FixedPointFree.lean]: (`MonoidHom` seems better than `MulHom` since the whole point is that `1` is special).

File: Mathlib/Data/List/BigOperators/Basic.lean
Comment (tb65536) [reply Mathlib/Data/List/BigOperators/Basic.lean]: I've shuffled things around to use `/`, but of course I still can't use `Finset.prod_range_div` directly.

**Comment by tb65536** [comment]
I've separated out some annoying `HomClass` refactors to https://github.com/leanprover-community/mathlib4/compare/tb_iterate_class

File: Mathlib/GroupTheory/FixedPointFree.lean
Comment (jcommelin) [inline Mathlib/GroupTheory/FixedPointFree.lean:69]:
```suggestion
theorem comm_of_involutive (g h : G) : g * h = h * g := by
```
Alternatively, consider using the `Commute` predicate.

**Comment by jcommelin** [comment]
A few more comments. But this should be good to go after those are addressed.

File: Mathlib/GroupTheory/FixedPointFree.lean
Comment (YaelDillies) [inline Mathlib/GroupTheory/FixedPointFree.lean:69]:
```suggestion
theorem commute_all_of_involutive (g h : G) : Commute g h := by
```
or something like that to match `Commute.all`?

File: Mathlib/Data/List/BigOperators/Basic.lean
Comment (YaelDillies) [inline Mathlib/Data/List/BigOperators/Basic.lean:465]:
```suggestion
@[to_additive "Cancellation of a telescoping sum."]
```

File: Mathlib/Data/List/BigOperators/Basic.lean
Comment (YaelDillies) [inline Mathlib/Data/List/BigOperators/Basic.lean:451]:
```suggestion
@[to_additive "Cancellation of a telescoping sum."]
```

File: Mathlib/GroupTheory/FixedPointFree.lean
Comment (YaelDillies) [inline Mathlib/GroupTheory/FixedPointFree.lean:69]:
```suggestion
theorem commute_all_of_involutive (g h : G) : Commute g h := by
```

File: Mathlib/GroupTheory/FixedPointFree.lean
Comment (YaelDillies) [inline Mathlib/GroupTheory/FixedPointFree.lean:75]:
```suggestion
def commGroupOfInvolutive : CommGroup G := .mk (hφ.commute_all_of_involutiveh2)
```

**Review by YaelDillies — APPROVED** [review]
Thanks!

maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**
